src/function/table/read_csv.cpp

```diff
@@ -30,78 +30,23 @@ static unique_ptr<FunctionData> ReadCSVBind(ClientContext &context, TableFunctio
-		} else if (loption == "header") {
-			options.header = BooleanValue::Get(kv.second);
-			options.has_header = true;
-		} else if (loption == "quote") {
-			options.quote = StringValue::Get(kv.second);
-			options.has_quote = true;
-		} else if (loption == "escape") {
-			options.escape = StringValue::Get(kv.second);
-			options.has_escape = true;
-		} else if (loption == "nullstr") {
-			options.null_str = StringValue::Get(kv.second);
-		} else if (loption == "sample_size") {
-			int64_t sample_size = kv.second.GetValue<int64_t>();
-			if (sample_size < 1 && sample_size != -1) {
-				throw BinderException("Unsupported parameter for SAMPLE_SIZE: cannot be smaller than 1");
-			}
-			if (sample_size == -1) {
-				options.sample_chunks = std::numeric_limits<uint64_t>::max();
-				options.sample_chunk_size = STANDARD_VECTOR_SIZE;
-			} else if (sample_size <= STANDARD_VECTOR_SIZE) {
-				options.sample_chunk_size = sample_size;
-				options.sample_chunks = 1;
-			} else {
-				options.sample_chunk_size = STANDARD_VECTOR_SIZE;
-				options.sample_chunks = sample_size / STANDARD_VECTOR_SIZE;
-			}
-		} else if (loption == "sample_chunk_size") {
-			options.sample_chunk_size = kv.second.GetValue<int64_t>();
-			if (options.sample_chunk_size > STANDARD_VECTOR_SIZE) {
-				throw BinderException(
-				    "Unsupported parameter for SAMPLE_CHUNK_SIZE: cannot be bigger than STANDARD_VECTOR_SIZE %d",
-				    STANDARD_VECTOR_SIZE);
-			} else if (options.sample_chunk_size < 1) {
-				throw BinderException("Unsupported parameter for SAMPLE_CHUNK_SIZE: cannot be smaller than 1");
-			}
-		} else if (loption == "sample_chunks") {
-			options.sample_chunks = kv.second.GetValue<int64_t>();
-			if (options.sample_chunks < 1) {
-				throw BinderException("Unsupported parameter for SAMPLE_CHUNKS: cannot be smaller than 1");
-			}
-		} else if (loption == "all_varchar") {
-			options.all_varchar = BooleanValue::Get(kv.second);
-		} else if (loption == "dateformat") {
-			options.has_format[LogicalTypeId::DATE] = true;
-			auto &date_format = options.date_format[LogicalTypeId::DATE];
-			date_format.format_specifier = StringValue::Get(kv.second);
-			string error = StrTimeFormat::ParseFormatSpecifier(date_format.format_specifier, date_format);
-			if (!error.empty()) {
-				throw InvalidInputException("Could not parse DATEFORMAT: %s", error.c_str());
-			}
-		} else if (loption == "timestampformat") {
-			options.has_format[LogicalTypeId::TIMESTAMP] = true;
-			auto &timestamp_format = options.date_format[LogicalTypeId::TIMESTAMP];
-			timestamp_format.format_specifier = StringValue::Get(kv.second);
-			string error = StrTimeFormat::ParseFormatSpecifier(timestamp_format.format_specifier, timestamp_format);
-			if (!error.empty()) {
-				throw InvalidInputException("Could not parse TIMESTAMPFORMAT: %s", error.c_str());
+		set.clear();
+		set.emplace_back(kv.second);
+		if (options.SetReadOption(loption, set, names)) {
+			continue;
+		}
+		if (loption == "columns") {
```

Comment:
If we check these special cases before calling `options.SetReadOption` we can immediately throw an exception in the `SetReadOption` function when something goes wrong.